url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/Simulation.java

```diff
@@ -0,0 +1,1042 @@
+						act.addShiftable(key, shiftable);
+						act.addConfig(key, exclusive);
+						// add appliances
+						appliances = actModApps.get(key);
+						String[] containsAppliances = new String[appliances.size()];
+						for(int l = 0; l < appliances.size(); l++) 
+							containsAppliances[l] = inst_id + "_" + appliances.get(l).getId();
+						act.addAppliances(containsAppliances, existing, key);
+					}
+					person.addActivity(act);
+//					act.setParentId(person_id);
+					String act_id = person_id + actid;
+					act.setId(act_id);
+				}
+				installations.add(inst);
+			}
+		}
+
+//		for (Installation inst: installations)
+//		{
+//			inst.printInstallationInfo();
+//		}
+
+	}
+
+	/**
+	 * Runs the simulation and outputs results to the csv files and the selected database.
+	 * Results include the expected, active and reactive power per installation, 
+	 * along with various KPIs that can be printed out using the  {@link #printKPIs() printKPIs} method.
+	 */
+	public void runSimulation () {
+		if(seed > 0) {
+			orng = new ORNG(seed);
+		} else {
+			orng = new ORNG();
+		}
+
+		try {
+			//  			DBObject objRun = DBConn.getConn().getCollection(MongoRuns.COL_RUNS).findOne(query);
+			System.out.println("Run " + dbname + " started @ " + Calendar.getInstance().getTime());
+			calculateExpectedPower(dbname);
+			//  			System.out.println("EP calculated");
+			long startTime = System.currentTimeMillis();
+			int mccount = 0;
+			double mcrunsRatio = 1.0/mcruns;
+			for(int i = 0; i < mcruns; i++) {
+				tick = 0;
+				double avgPPowerPerHour = 0;
+				double avgQPowerPerHour = 0;
+				double[] avgPPowerPerHourPerInst = new double[installations.size()];
+				double[] avgQPowerPerHourPerInst = new double[installations.size()];
+				double maxPower = 0;
+				//  	  			double cycleMaxPower = 0;
+				double avgPower = 0;
+				double energy = 0;
+				double energyOffpeak = 0;
+				double cost = 0;
+				//  	  			double billingCycleEnergy = 0;
+				//  	  			double billingCycleEnergyOffpeak = 0;
+				while (tick < endTick) {
+					// If it is the beginning of the day create the events
+					if (tick % Constants.MIN_IN_DAY == 0) {
+						//  	  				System.out.println("Day " + ((tick / Constants.MIN_IN_DAY) + 1));
+						for (Installation installation: installations) {
+							//  						System.out.println("Installation: " + installation.getName());
+							installation.updateDailySchedule(tick, queue, simulationWorld.getResponseType(), orng);
+
+						}
+						//  					System.out.println("Daily queue size: " + queue.size() + "(" + 
+						//  					simulationWorld.getSimCalendar().isWeekend(tick) + ")");
+					}
+					Event top = queue.peek();
+					while (top != null && top.getTick() == tick) {
+						Event e = queue.poll();
+						boolean applied = e.apply();
+						if(applied) {
+							if(e.getAction() == Event.SWITCH_ON) {
+								try {
+									//m.addOpenTick(e.getAppliance().getId(), tick);
+								} catch (Exception exc) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)